assets/chef-recipes/cookbooks/suse-connect/libraries/registration.rb

```diff
@@ -24,16 +24,32 @@ def self.load_credentials
       # Generate registration command for the machine
       # @param suse_connect_parameters [Hash] subscription parameters in format {'email' : String, 'key' : String, 'registration_proxy_url' : String }
       def self.register_node_command(suse_connect_parameters)
-        # Copying configuration script from the registration proxy requires HTTP protocol
-        http_proxy_url = suse_connect_parameters['registration_proxy_url'].sub('https://', 'http://')
-        "curl #{URI.join(http_proxy_url, '/tools/rmt-client-setup')} --output rmt-client-setup; yes | sh rmt-client-setup #{suse_connect_parameters['registration_proxy_url']}; SUSEConnect -r #{suse_connect_parameters['key']} -e #{suse_connect_parameters['email']} --url #{suse_connect_parameters['registration_proxy_url']}"
-      end
+        if use_proxy?(suse_connect_parameters)
+          # Copying configuration script from the registration proxy requires HTTP protocol
+          http_proxy_url = suse_connect_parameters['registration_proxy_url'].sub('https://', 'http://')
+          return "curl #{URI.join(http_proxy_url, '/tools/rmt-client-setup')} --output rmt-client-setup; yes | sh rmt-client-setup #{suse_connect_parameters['registration_proxy_url']}; SUSEConnect -r #{suse_connect_parameters['key']} -e #{suse_connect_parameters['email']} --url #{suse_connect_parameters['registration_proxy_url']}"
+        end
 
+        "SUSEConnect -r #{suse_connect_parameters['key']} -e #{suse_connect_parameters['email']} --url https://scc.suse.com"
+      end
       # Generate deregistration command for the machine
       # @param credentials [Hash] machine credentials in format {'username' : String, 'password' : String }
       # @param registration_proxy_url [String] URL for SUSE Registration Proxy server
-      def self.deregister_node_command(registration_proxy_url, credentials)
-        "curl -k -X DELETE -u #{credentials['username']}:#{credentials['password']} #{URI.join(registration_proxy_url, REGISTRATION_ENDPOINT)}"
+      def self.deregister_node_command(suse_connect_parameters, credentials)
+        registration_url = use_proxy?(suse_connect_parameters) ? suse_connect_parameters['registration_proxy_url'] : 'https://scc.suse.com'
+        "curl -k -X DELETE -u #{credentials['username']}:#{credentials['password']} #{URI.join(registration_url, REGISTRATION_ENDPOINT)}"
+      end
+
+      def self.proxy_available?(registration_proxy_url)
+        cmd = Mixlib::ShellOut.new("curl -k #{registration_proxy_url} --connect-timeout 2")
+        cmd.run_command
+        !cmd.error?
+      end
+
+      # Check if the system can be registered via the Registration Proxy Server
+      # @param suse_connect_parameters [Hash] subscription parameters in format {'provider': String, 'email' : String, 'key' : String, 'registration_proxy_url' : String }
+      def self.use_proxy?(suse_connect_parameters)
```

Comment:
Consider to cache this value inside the module constant, so it will not try to connect to server several times.